src/main/java/nl/tudelft/jpacman/level/Player.java

```diff
@@ -34,6 +34,11 @@
      */
     private boolean alive;
 
+    /**
+     * <code>Ghost</code> iff this player died by collision.
+     */
+    private Unit killer;
```

Comment:
I believe the reason for adding this method was determining the cause of death. This could be after collision with a ghost, however, this could also be after the malicious code killed the player (`setAlive(false)`). If we'd get rid of the alive variable it's no longer possible to differentiate these two situations. For normal use however, I can see that the current solution is not necessarily the best one. Not sure what's the best way to solve this 😕. Any ideas, maybe?

We 'know' that it's a ghost. However, I can imagine that in some extension you might want to have another 'killing' unit (maybe some dangerous pellets or something). That's why we went with `Unit`. I see the javadoc does not reflect this though.